extensions/amp-font/0.1/amp-font.js

```diff
@@ -0,0 +1,316 @@
+    });
+    this.document_.body.appendChild(containerElement);
+    DEFAULT_FONTS_.forEach(font => {
+      let normalizedFont = normalizeFontFamily_(font);
+      let defaultFontElementId =
+          '-amp-font-' + ncustomFontFamily_ + '-' + normalizedFont;
+      let element = this.document_.getElementById(defaultFontElementId);
+      this.measurements_[defaultFontElementId] = {
+        width: element./*OK*/offsetWidth,
+        height: element./*OK*/offsetHeight
+      };
+    });
+  }
+
+
+  /**
+   * Make the browser that don't support font loading events to download the
+   * custom font by creating an element (with text) not visible on the viewport.
+   * font download is detected by comparing the elements height and width with
+   * measurements taken in step1.
+   * @private
+   */
+  forceLoadFont_() {
+    var  hasWidthChanged = false;
+    var hasHeightChanged = false;
+    var ncustomFontFamily_ = normalizeFontFamily_(this.customFontFamily_);
+    DEFAULT_FONTS_.some(font => {
+      if(this.hasTimedOut_) {
+        this.onFontLoadError_();
+        return true;
+      }
+      let normalizedFont = normalizeFontFamily_(font);
+      let defaultFontElementId =
+          '-amp-font-' + ncustomFontFamily_ + '-' + normalizedFont;
+      let element = this.document_.getElementById(defaultFontElementId);
+      let currentFontFamily = element.style.fontFamily;
+      if (currentFontFamily.indexOf(this.customFontFamily_) == -1) {
+        element.style.fontFamily =
+            this.customFontFamily_ + ',' + currentFontFamily;
+      }
+      var defaultDimensions =
+          this.measurements_[defaultFontElementId];
+       hasWidthChanged = (
+        Math.abs(
+          defaultDimensions.width - element./*OK*/offsetWidth) > TOLERANCE_);
+      hasHeightChanged = (
+        Math.abs(
+          defaultDimensions.height - element./*OK*/offsetHeight) > TOLERANCE_);
+      if ( hasWidthChanged || hasHeightChanged) {
+        return true;
+      }
+    });
+    if ( hasWidthChanged || hasHeightChanged) {
+      this.onFontLoadSuccess_();
+    } else if(this.hasTimedOut_) {
+      this.onFontLoadError_();
+    } else {
+      RAF_(() => this.forceLoadFont_());
+    }
+  }
+
+
+  /**
+   * @private
+   */
+  onFontLoadSuccess_() {
+    var addClassName = this.element.getAttribute('on-load-class') || '';
+    this.documentElement_.classList.add(addClassName);
+    this.dispose_();
+  }
+
+
+  /**
+   * @private
+   */
+  onFontLoadError_() {
+    var addClassName = this.element.getAttribute('on-error-class') || '';
+    this.documentElement_.classList.add(addClassName);
+    this.dispose_();
+  }
+
+
+  /**
+   * @private
+   */
+  dispose_() {
+    var removeClassName =
+        this.element.getAttribute('while-loading-class') || '';
+    this.documentElement_.classList.remove(removeClassName);
+    this.measurements_ = Object.create(null);
+    let containerElement = this.document_.getElementById(
```

Comment:
Likewise here: I'd just store the created element in an instance var and avoid `getElementById`